src/field_theory/ratfunc.lean

```diff
@@ -1095,6 +1095,40 @@ begin
   { exact algebra_map_ne_zero (denom_ne_zero y) },
 end
 
+lemma nat_degree_well_defined {r₁ r₂ s₁ s₂ : polynomial K} (hr₁ : r₁ ≠ 0) (hs₁ : s₁ ≠ 0)
+  (hr₂ : r₂ ≠ 0) (hs₂ : s₂ ≠ 0) (h_eq : r₁*s₂ = r₂*s₁) :
+  (r₁.nat_degree : ℤ) - s₁.nat_degree = (r₂.nat_degree : ℤ) - s₂.nat_degree :=
+begin
+  rw sub_eq_sub_iff_add_eq_add,
+  norm_cast,
+  rw [← nat_degree_mul hr₁ hs₂, ← nat_degree_mul hr₂ hs₁, h_eq]
+end
+
+lemma num_mul_denom_add_denom_mul_num_ne_zero {x y : ratfunc K} (hxy : x + y ≠ 0) :
+  x.num * y.denom + x.denom * y.num ≠ 0 :=
+begin
+  intro h_zero,
+  have h := num_denom_add x y,
+  rw [h_zero, zero_mul] at h,
+  exact (mul_ne_zero (num_ne_zero hxy) (mul_ne_zero x.denom_ne_zero y.denom_ne_zero)) h
+end
+
+lemma nat_degree_add_eq_nat_degree_num_denom_add {x y : ratfunc K} (hxy : x + y ≠ 0) :
```

Comment:
I don't understand the name here, you should mention `sub` I think.

Reply:
This could be named `degree_add_eq_nat_degree_num_add_sub_nat_degree_denom_add`, but isn't this too long?.

Reply:
I renamed it to `int_degree_add_eq_nat_degree_num_add_sub_nat_degree_denom_add`